react/features/base/participants/components/ParticipantView.native.js

```diff
@@ -217,14 +217,14 @@ class ParticipantView extends Component<Props> {
                     === JitsiParticipantConnectionStatus.ACTIVE)
                 && shouldRenderVideoTrack(videoTrack, waitForVideoStarted);
 
-        // Is the avatar to be rendered?
-        let renderAvatar = Boolean(!renderVideo && avatar);
-
         // The consumer of this ParticipantView is allowed to forbid showing the
         // video if the private logic of this ParticipantView determines that
         // the video could be rendered.
         renderVideo = renderVideo && _toBoolean(this.props.showVideo, true);
 
+        // Is the avatar to be rendered?
+        let renderAvatar = Boolean(!renderVideo && avatar);
+
```

Comment:
Could you please elaborate the need for this change? The initial intent of the source code was to determine whether video or avatar was to be rendered and then allow the consumer of the React Component `ParticipantView` control over whether really video and avatar are to be rendered (because the consumer is knows best about their use case).

I presume it's necessary for the new source code in this PR in `Thumbnail` but I'd appreciate comments in the source code explaining the repurposed React Component prop `showVideo`.

Reply:
The reason for this change was that the renderAvatar value didn't take the props.showVideo into account, only the earlier checks (line 214), but it didn't cause an issue yet, because the showVideo and the showAvatar props always had the same value (see usage in Thumbnail and LargeVideo). But now that they can have different values (like the new usage in Thumbnail) this actually causes a bug. I don't want to add this long explanation into the code, but I surely add something to make it obvious.